tt.go

```diff
@@ -0,0 +1,278 @@
+
+	// MakeTypeToUnsignedMapper creates a mapper from Type to unsigned.
+	MakeTypeToUnsignedMapper() Mapper[Type, any]
+
+	// MakeTypeToDatetimeMapper creates a mapper from Type to datetime.
+	MakeTypeToDatetimeMapper() Mapper[Type, any]
+
+	// MakeTypeToUUIDMapper creates a mapper from Type to uuid.
+	MakeTypeToUUIDMapper() Mapper[Type, any]
+
+	// MakeTypeToMapMapper creates a mapper from Type to map.
+	MakeTypeToMapMapper() Mapper[Type, any]
+
+	// MakeTypeToArrayMapper creates a mapper from Type to array.
+	MakeTypeToArrayMapper() Mapper[Type, any]
+
+	// MakeTypeToVarbinaryMapper creates a mapper from Type to varbinary.
+	MakeTypeToVarbinaryMapper() Mapper[Type, any]
+
+	// MakeTypeToDoubleMapper creates a mapper from Type to double.
+	MakeTypeToDoubleMapper() Mapper[Type, any]
+
+	// MakeTypeToDecimalMapper creates a mapper from Type to decimal.
+	MakeTypeToDecimalMapper() Mapper[Type, any]
+
+	// MakeTypeToIntegerMapper creates a mapper from Type to integer.
+	MakeTypeToIntegerMapper() Mapper[Type, any]
+
+	// MakeTypeToNumberMapper creates a mapper from Type to number.
+	MakeTypeToNumberMapper() Mapper[Type, any]
+
+	// MakeTypeToAnyMapper creates a mapper from Type to any.
+	MakeTypeToAnyMapper() Mapper[Type, any]
+
+	// MakeTypeToScalarMapper creates a mapper from Type to scalar.
+	MakeTypeToScalarMapper() Mapper[Type, any]
+
+	// MakeTypeToIntervalMapper creates a mapper from Type to interval.
+	MakeTypeToIntervalMapper() Mapper[Type, any]
+
+	// MakeNullableMapper extends the incoming mapper to a nullable mapper.
+	MakeNullableMapper(Mapper[Type, any]) Mapper[Type, any]
+}
+
+// StringToTTMapperOpts are options of mappers from string to tarantool types.
+type StringToTTMapperOpts struct {
+	// ThSeparators are thousands separators for numeric types.
+	// In fact, these symbols will be ignored for numeric types.
+	// For example:
+	// ThSeparators = "`#e": 1`2`3 -> 123, 123#456#789 -> 123456789, 1e5 -> 15.
+	ThSeparators string
+	// DecSeparators are additional decimal separators for numeric types (besides `.`).
+	// In fact, these symbols will be replaced with `.`.
+	// For example:
+	// DecSeparators = ",#": 12,3 -> 12.3, 100.500 -> 100.500, 12#13 -> 12.13.
+	DecSeparators string
+	// NullValue is a value that is interpreted as null.
+	NullValue string
+}
+
+// GetDefaultStringToTTMapperOpts returns default string to tt mappers options.
+func GetDefaultStringToTTMapperOpts() StringToTTMapperOpts {
+	return StringToTTMapperOpts{
+		ThSeparators:  defaultThSeparators,
+		DecSeparators: defaultDecSeparators,
+		NullValue:     defaultNullValue,
+	}
+}
+
+// StringToTTMapperFactory is the default TypeToTTMapperFactory for strings.
+// To customize the creation of mappers, inherit from it and override the necessary methods.
+type StringToTTMapperFactory struct {
+	opts StringToTTMapperOpts
+}
+
+// NewStringToTTMapperFactory creates StringToTTMapperFactory.
+func NewStringToTTMapperFactory() *StringToTTMapperFactory {
```

Comment:
What code should I write to create this factory with custom dec separator?

Reply:
```golang
facture := tupleconv.NewStringToTTMapperFactory()
facture.Cfg(tupleconv.StringToTTMapperOpts{DecSeparators: ",."})
```

Reply:
> ```go
> facture := tupleconv.NewStringToTTMapperFactory()
> facture.Cfg(tupleconv.StringToTTMapperOpts{DecSeparators: ",."})
> ```

Can we avoid new external struct with public members and use functional options to configure the factory?

Reply:
Functional options added.